File: eShopModernizedSpringSolution/src/main/java/com/eshop/modernized/dto/PaginatedItemsViewModel.java
package com.eshop.modernized.dto;

import java.util.List;

public record PaginatedItemsViewModel<T>(
        int pageIndex,
        int pageSize,
        long count,
        List<T> data
) {

    public int totalPages() {
        return (int) Math.ceil((double) count / pageSize);
    }
}

File: eShopModernized/src/eShop.Catalog.Domain/PaginatedItemsViewModel.cs
namespace eShop.Catalog.Domain;

/// <summary>
/// Paging envelope ported from the legacy MVC/Web Forms ViewModel of the same name.
/// </summary>
public class PaginatedItemsViewModel<TEntity>
    where TEntity : class
{
    public PaginatedItemsViewModel(int pageIndex, int pageSize, long count, IEnumerable<TEntity> data)
    {
        ActualPage = pageIndex;
        ItemsPerPage = pageSize;
        TotalItems = count;
        TotalPages = (int)Math.Ceiling((decimal)count / pageSize);
        Data = data;
    }

    public int ActualPage { get; private set; }

    public int ItemsPerPage { get; private set; }

    public long TotalItems { get; private set; }

    public int TotalPages { get; set; }

    public IEnumerable<TEntity> Data { get; private set; }
}
